presto-main/src/main/java/com/facebook/presto/server/BasicQueryInfo.java

```diff
@@ -88,6 +97,36 @@ public BasicQueryInfo(
         this.warnings = requireNonNull(warnings, "warnings is null");
     }
 
+    public BasicQueryInfo(
+            QueryId queryId,
+            SessionRepresentation session,
+            Optional<ResourceGroupId> resourceGroupId,
+            QueryState state,
+            MemoryPoolId memoryPool,
+            boolean scheduled,
+            URI self,
+            String query,
+            BasicQueryStats queryStats,
+            ExecutionFailureInfo failureInfo,
+            Optional<QueryType> queryType,
+            List<PrestoWarning> warnings)
+    {
+        this(
+                queryId,
+                session,
+                resourceGroupId,
+                state,
+                memoryPool,
+                scheduled,
+                self,
+                query,
+                queryStats,
+                failureInfo != null && failureInfo.getErrorCode() != null ? failureInfo.getErrorCode().getType() : null,
```

Comment:
Not sure if it worths adding parenthesis to make it clearer :)

`(failureInfo != null && failureInfo.getErrorCode() != null) ? failureInfo.getErrorCode().getType() : null`